BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/DistanceUtil.java

```diff
@@ -21,4 +27,25 @@ public static int getMaximumEvenIncrement(double distance) {
         return (int) Math.pow(10, Math.floor(Math.log10(distance)));
     }
 
+    public static Location walkingSpeedFilter(Location oldLocation, Location newLocation) {
+        return speedFilter(oldLocation, newLocation, HUMAN_WALKING_SPEED);
+    }
+
+    /**
+     * Define a maximum movement speed to restrain a new location being further away than the
+     * distance possible in that time window.
+     */
+    public static Location speedFilter(Location oldLocation, Location newLocation, double maximumSpeed) {
+        double distance = oldLocation.getDistanceTo(newLocation);
+        long timestampDelta = newLocation.getTimestamp() - oldLocation.getTimestamp();
+        double currentSpeed = distance / ((float) timestampDelta / TimeUnit.SECONDS.toMillis(1));
```

Comment:
`ArithmeticException` for division by 0 if the locations have the same timestamp.